index.html

```diff
@@ -9064,8 +9050,454 @@
+}
+
+function render_enhanced_chat_instruct(input)
+{
+	const contextDict = { sysOpen: '<sys_context_koboldlite_internal>', youOpen: '<user_context_koboldlite_internal>', AIOpen: '<AI_context_koboldlite_internal>', closeTag: '<end_of_context_koboldlite_internal>' }
+	let you = get_instruct_starttag(); let bot = get_instruct_endtag(); // Instruct tags will be used to wrap text in styled bubbles.
+	let as = aestheticInstructUISettings;
+	
+	// We'll transform the input to a well-formatted HTML string. First, trim any newline characters after seq_start and seq_end.
+	let newbodystr = (input.startsWith(you) || input.startsWith(bot)) ? input : style('sys') + input;
+	newbodystr = newbodystr.replaceAll(you + '\n', you).replaceAll(bot + '\n', bot).replaceAll(you + ' ', you).replaceAll(bot + ' ', bot).replaceAll('"', '&quot;');
+	newbodystr = newbodystr.replaceAll(you, style('you')).replaceAll(bot, style('AI')) + contextDict.closeTag; // Style incoming messages and outcoming messages appropriately.
+	if (aestheticInstructUISettings.use_markdown) {
+		for (let role of aestheticTextStyleRoles) {
+			newbodystr = newbodystr.replace(new RegExp(`${contextDict[`${role}Open`]}([^]*?)${contextDict.closeTag}`, 'g'), (match, p1) => {
+				let styleRole = aestheticInstructUISettings.use_uniform_colors ? 'uniform' : role;
+				return match.replace(/\*(\S[^*]+\S)\*/g, wrapperSpan(styleRole, 'action')).replace(/&quot;(.*?)&quot;/g, wrapperSpan(styleRole, 'speech'));
+			});
+		}
+		newbodystr = applyStylizedCodeBlocks();
+	}
+	return (newbodystr + "</p></div></div>\n").replaceAll('\r\n','<br>').replaceAll('\n','<br>');	// Stylize, handle newlines, and return the outcome string.
+
+	// Helper functions to allow styling properly.
+	function image(role) {
+		const portraitSrc = localStorage.getItem(`${role}_portrait`) || as[`${role}_portrait`];
+		return as.border_style == 'None' || role == 'sys' ? '' : `<img src='${portraitSrc}' style='width:${as.portraitSize().width}px; height:${as.portraitSize().height}px; margin: 10px 0px; border-radius: ${as.portraitRadius()};'>`;
+	}
+	function style(role) { return `${contextDict.closeTag}</p></div></div><div style='display:flex; align-items:stretch; flex-direction: row;'>${image(role)}<div style='flex: 1; color: ${as[`text_color_${aestheticInstructUISettings.use_uniform_colors ? 'uniform' : role}`].color}; background-color:${as[`bubbleColor_${role}`]}; padding: ${as.padding()}; margin: ${as.margin()}; min-height:${as.background_minHeight}px; font-size: ${as.font_size}px; display: flex; flex-direction:column; align-items: flex-start; justify-content: center; border-radius: 15px'><p>${contextDict[`${role}Open`]}`; }
+	function wrapperSpan(role, type) { let textStyle = as[`${type}_color_${role}`]; return `<span style='color: ${textStyle.color}; font-style: ${textStyle.fontStyle}; font-weight: ${textStyle.fontWeight}'>$1</span>`; }
+
+	function applyStylizedCodeBlocks() {
+		let blocks = newbodystr.split(/(```.*?\n[\s\S]*?```)/gs);
```

Comment:
actually missed this in the previous review, but the dot matches singleline regex flag `s` also needs to be polyfilled due to compatibility reasons, as firefox support for it was added much later (v80+). 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/dotAll

let me know if you have trouble modifying the regex